orderbook/usecase/orderbook_usecase_test.go

```diff
@@ -221,6 +223,184 @@ func (s *OrderbookUsecaseTestSuite) TestProcessPool() {
+				return context.Background()
+			},
+			setupMocks: func(usecase *orderbookusecase.OrderbookUseCaseImpl, orderbookrepository *mocks.OrderbookRepositoryMock, grpcclient *mocks.OrderbookGRPCClientMock, poolsUsecase *mocks.PoolsUsecaseMock, tokensusecase *mocks.TokensUsecaseMock) {
+				poolsUsecase.GetAllCanonicalOrderbookPoolIDsFunc = func() ([]domain.CanonicalOrderBooksResult, error) {
+					return nil, assert.AnError
+				}
+			},
+			expectedError: &types.FailedGetAllCanonicalOrderbookPoolIDsError{},
+		},
+		{
+			name: "context is done before processing all orderbooks",
+			setupContext: func() context.Context {
+				ctx, cancel := context.WithCancel(context.Background())
+				cancel()
+				return ctx
+			},
+			setupMocks: func(usecase *orderbookusecase.OrderbookUseCaseImpl, orderbookrepository *mocks.OrderbookRepositoryMock, grpcclient *mocks.OrderbookGRPCClientMock, poolsUsecase *mocks.PoolsUsecaseMock, tokensusecase *mocks.TokensUsecaseMock) {
+				withGetAllCanonicalOrderbookPoolIDs(poolsUsecase)
+			},
+			expectedError: context.Canceled,
+		},
+		{
+			name: "processOrderBookActiveOrders returns an error",
+			setupContext: func() context.Context {
+				return context.Background()
+			},
+			setupMocks: func(usecase *orderbookusecase.OrderbookUseCaseImpl, orderbookrepository *mocks.OrderbookRepositoryMock, grpcclient *mocks.OrderbookGRPCClientMock, poolsUsecase *mocks.PoolsUsecaseMock, tokensusecase *mocks.TokensUsecaseMock) {
+				withGetAllCanonicalOrderbookPoolIDs(poolsUsecase)
+
+				grpcclient.MockGetActiveOrdersCb = func(ctx context.Context, contractAddress string, ownerAddress string) (orderbookdomain.Orders, uint64, error) {
+					return nil, 0, assert.AnError
+				}
+			},
+			expectedError:        &types.FailedProcessingOrderbookActiveOrdersError{},
+			expectedIsBestEffort: false,
+		},
+		{
+			name: "isBestEffort set to true when one orderbook is processed with best effort",
+			setupContext: func() context.Context {
+				return context.Background()
+			},
+			setupMocks: func(usecase *orderbookusecase.OrderbookUseCaseImpl, orderbookrepository *mocks.OrderbookRepositoryMock, grpcclient *mocks.OrderbookGRPCClientMock, poolsUsecase *mocks.PoolsUsecaseMock, tokensusecase *mocks.TokensUsecaseMock) {
+				withGetAllCanonicalOrderbookPoolIDs(poolsUsecase)
+
+				grpcclient.MockGetActiveOrdersCb = func(ctx context.Context, contractAddress string, ownerAddress string) (orderbookdomain.Orders, uint64, error) {
+					return orderbookdomain.Orders{
+						{
+							OrderId: 1,
+						},
+					}, 1, nil
+
+				}
+
+				withGetMetadataByChainDenom(tokensusecase)
+
+				// Set is best effort to true
+				orderbookrepository.GetTickByIDFunc = func(poolID uint64, tickID int64) (orderbookdomain.OrderbookTick, bool) {
+					return orderbookdomain.OrderbookTick{}, false
+				}
+			},
+			expectedError:        nil,
+			expectedIsBestEffort: true,
+		},
+		{
+			name: "successful retrieval of active orders",
```

Comment:
I think that we would benefit from more cases:
1. 3 orders returned. 1 from order book A, 2 from orderbook B -> 3 orders are returned as intended
2. 2 orders returned. 1 from order book A, 1 from order book B. Order book B is not canonical -> only 1 order is returned
3. 2 orders returned. 1 is not from the input address -> 1 order returned

This would require defining more orders.

What we can do is create 1 `defaultOrder` at the top of the test. Then, use a decorator pattern to change them as needed for readability. For example:


```go
withOrderID := func(o Order, poolID uint64) Order {
   o.PoolID = poolID
   return o
}
```

Reply:
I have implemented as per your suggestion! Just one question, by `canonical` you meant it's not valid or something else? For the rest case I was not able to implement that because there is no way to check does the order belongs to the input address since order does contain address field.

Reply:
> by canonical you meant it's not valid or something else?

Sorry for the confusion - I meant that there is no order in that order book for a given user